[Workers and Pages] Smart placement updates (#8856)
cloudflare/cloudflare-docs: content/pages/platform/functions/smart-placement.md

---
pcx_content_type: concept
title: Smart Placement (beta)
---

{{<beta>}}Smart Placement{{</beta>}}

Smart Placement (beta) automatically places your workloads in an optimal location that minimizes latency and speeds up your applications. By default, [Workers](/workers/) and [Pages Functions](/pages/platform/functions/) are invoked in a data center closest to where the request was received. If you are running back-end logic in a Pages Function, it may be more performant to run that Worker closer to your back-end infrastructure rather than the end user. 

You may benefit from Smart Placement if you are making multiple round trips to a centralized database, API or origin server in a Pages Function.

## Background

> Smart Placement applies to Pages Functions and middleware. Assets will be served globally. What else should pages users know?

Author: @sidharthachatterjee can you add content here + any best practices for Pages users below

Reviewer: On it!

> Smart Placement applies to Pages Functions and middleware. Assets will be served globally. What else should pages users know?

Reviewer: ```suggestion
Smart Placement applies to Pages Functions and middleware. Assets are always served globally and closest to your users. 

Smart Placement on Pages currently has some caveats. While assets are always meant to be served from a location closest to the user, there are two exceptions to this:

- If using middleware when Smart Placement is enabled, all assets will be served from a location closest to your back-end infrastructure. This may result in an unexpected increase in latency as a result. 

- When using [`env.ASSETS.fetch`](https://developers.cloudflare.com/pages/platform/functions/advanced-mode/), assets served via the ASSETS fetcher from your Pages Function are served from the same location as your Function. This could be the location closest to your back-end infrastructure and _not_ the user. 

```

{{<Aside type= "note">}}

For more background and to understand how Smart Placement works visit [Smart Placement](/workers/platform/smart-placement/)

> {{</Aside>}}
> 
> 
> 
> ## Enable Smart Placement (beta)

Reviewer 2: ```suggestion
{{</Aside>}}

## Enable Smart Placement (beta)
```

> Smart Placement is available users on all plans. 

Reviewer 2: ```suggestion
Smart Placement is available on all plans. 
```

> ### Enable Smart Placement on the dashboard:
> 1. Log in to the Cloudflare dashboard.
> 2. In Account Home, select Pages.
> 3. Select a Pages project
> 4. Select Settings > Functions.
> 5. Under 'Placement' enable 'Smart' mode 
> 6. . Send some initial traffic (approximately 20-30 requests) to your Worker. It takes a few minutes after you have sent traffic to your Worker for Smart Placement to take effect. 
> 7. View [request duration metrics](/workers/learning/metrics-and-analytics) about your Function under Functions Metrics.
> 
> 
> 
> ## Best practices

Reviewer 2: ```suggestion
### Enable Smart Placement on the dashboard:
1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com) and select your account.
2. In Account Home, select **Workers & Pages**.
3. Select your Pages project.
4. Select **Settings** > **Functions**.
5. Under **Placement**, enable **Smart** mode.
6. Send some initial traffic (approximately 20-30 requests) to your Pages Functions. It takes a few minutes after you have sent traffic to your Pages Function for Smart Placement to take effect. 
7. View your Function's [request duration metrics](/workers/learning/metrics-and-analytics) under Functions Metrics.

## Best practices
```
